From 949ba26de394a0b2a56274970a429316026bad99 Mon Sep 17 00:00:00 2001
From: Kever Yang <kever.yang@rock-chips.com>
Date: Thu, 4 Feb 2021 14:31:35 +0800
Subject: [PATCH] cpufreq: fix redefinition of 'rockchip_cpufreq_adjust_target'

No need to define rockchip_cpufreq_adjust_target in cpufreq.h for ARCH
other than rockchip.

Signed-off-by: Kever Yang <kever.yang@rock-chips.com>
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
Change-Id: I32295390f39c53dd1f320362c71f762efbb8e83e
---
 drivers/cpufreq/cpufreq.c | 4 ++++
 include/linux/cpufreq.h   | 9 ---------
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 4f56d241ca52..d568a099e865 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -33,6 +33,10 @@
 #include <linux/tick.h>
 #include <trace/events/power.h>
 
+#if IS_ENABLED(CONFIG_ARM_ROCKCHIP_CPUFREQ)
+#include "rockchip-cpufreq.h"
+#endif
+
 static LIST_HEAD(cpufreq_policy_list);
 
 static inline bool policy_is_inactive(struct cpufreq_policy *policy)
diff --git a/include/linux/cpufreq.h b/include/linux/cpufreq.h
index d02619286e9b..4c7068afa597 100644
--- a/include/linux/cpufreq.h
+++ b/include/linux/cpufreq.h
@@ -975,13 +975,4 @@ unsigned int cpufreq_generic_get(unsigned int cpu);
 int cpufreq_generic_init(struct cpufreq_policy *policy,
 		struct cpufreq_frequency_table *table,
 		unsigned int transition_latency);
-#ifdef CONFIG_ARM_ROCKCHIP_CPUFREQ
-unsigned int rockchip_cpufreq_adjust_target(int cpu, unsigned int freq);
-#else
-static inline unsigned int rockchip_cpufreq_adjust_target(int cpu,
-							  unsigned int freq)
-{
-	return freq;
-}
-#endif
 #endif /* _LINUX_CPUFREQ_H */
-- 
2.35.3

